### ISSUE TITLE
Loading both structures/trajectories with multiple-frame .gro files, instead of having to also load two things separately in MDAnalysis.Universe

### ISSUE THREAD

**Issue by sdtruong** [MDAnalysis/mdanalysis#3034]
It'd be helpful if MDAnalysis had a feature to read GROMACS .gro files with multiple frames and obtain trajectories from this. Currently, an accompanying trajectory .xtc file is needed, but VMD is able to visualize multiple-frame .gro files, for example.

**Comment by IAlibay** [comment]
I'm going to link to the original discussion entry here: https://groups.google.com/g/mdnalysis-discussion/c/-NkpWa5m-W4/m/AJstu-_OCQAJ

TLDR; the current alternative is probably to use the chemfiles interface, but an enhancement to the native reader would be good 👍

**Comment by lilyminium** [comment]
The spec is [here](http://manual.gromacs.org/documentation/current/reference-manual/file-formats.html#gro) for people like me who didn't realise it could be multi frame. It seems to work like pdb essentially. In the meantime, could you convert into a PDB format which does support trajectories @sdtruong?

**Comment by sdtruong** [comment]
> The spec is [here](http://manual.gromacs.org/documentation/current/reference-manual/file-formats.html#gro) for people like me who didn't realise it could be multi frame. It seems to work like pdb essentially. In the meantime, could you convert into a PDB format which does support trajectories @sdtruong?

That's a good suggestion. Unfortunately, when I converted from GRO to PDB format, MDAnalysis.Universe still only takes the first frame. When I print out the trajectory, I get 1 frame (e.g. len(u.trajectory) is 1).

**Comment by IAlibay** [comment]
@sdtruong MDAnalysis should be able to read multiframe PDB files. 

Can you provide extra information on how you converted the file?

Also could you confirm that the multi-PDB file has MODEL and ENDMDL entries between each frame?

**Comment by sdtruong** [comment]
> @sdtruong MDAnalysis should be able to read multiframe PDB files.
> 
> Can you provide extra information on how you converted the file?
> 
> Also could you confirm that the multi-PDB file has MODEL and ENDMDL entries between each frame?

Okay, I think I see the issue then. I tried to convert the .gro file to .pdb using GROMCAS ediconf command. When I did this, I only used ediconf -f file.gro -o file.pdb, but that only gave one frame. Is there a way to output multiple-frame .pdb files using GROMACS editconf? Sorry if this is outside the scope of this question.

**Comment by IAlibay** [comment]
@sdtruong not that I'm aware using editconf, trjconv should probably work though.

**Comment by sdtruong** [comment]
> @sdtruong not that I'm aware using editconf, trjconv should probably work though.

I'm looking at the trjconv documentation, and it doesn't look like it takes in .gro files as input (I also tried doing this to no avail). Correct me if I'm wrong, but doesn't trjconv only takes trajectory files? In that case, we're at square one.

**Comment by IAlibay** [comment]
Specifically looking at the 2020 documentation unless I'm misreading `gro` is a valid trajectory type: http://manual.gromacs.org/2020/onlinehelp/gmx-trjconv.html

(see under -f flag), if for some reason this does not work I'm afraid it will have to be taken up with the gmx devs.

The chemfiles alternative should work though.

**Comment by sdtruong** [comment]
> Specifically looking at the 2020 documentation unless I'm misreading `gro` is a valid trajectory type: http://manual.gromacs.org/2020/onlinehelp/gmx-trjconv.html
> 
> (see under -f flag), if for some reason this does not work I'm afraid it will have to be taken up with the gmx devs.
> 
> The chemfiles alternative should work for reading multi frame gro files though.

Yes, this workaround seems to have worked! I'm going to do this instead in the time being. For posterity, the workaround is this:

Multiple-frame .gro file = `file.gro`

Since you cannot just load all the frames using MDAnalysis.Universe, keep the .gro file but generate a new .xtc file. You can do this with trjconv. Something like this:

`gmx trjconv -f file.gro -o file.xtc`

Now, you can load the structure with the trajectory using `MDAnalysis.Universe('file.gro', 'file.xtc')`.

**Comment by Luthaf** [comment]
As mentioned by @IAlibay, the chemfiles interface in MDAnalysis can do this without converting to xtc first with `MDAnalysis.Universe("file.gro", "file.gro", format="CHEMFILES")`.
